lua/neogit/lib/input.lua

```diff
@@ -0,0 +1,63 @@
+local M = {}
+
+if not _G.neogit then _G.neogit = {} end
+if not _G.neogit.completers then _G.neogit.completers = {} end
+
+local function user_input_prompt(prompt, default_value, completion_function)
+  vim.fn.inputsave()
+
+  local args = {
+    prompt = prompt
+  }
+  if default_value then args.default = default_value end
+  if completion_function then args.completion = 'customlist,v:lua.neogit.completers.'..completion_function end
+
+  local result = vim.fn.input(args)
+
+  vim.fn.inputrestore()
+  return result
+end
+
+local chars = vim.split('abcdefghijklmnopqrstuvwxyzABCDEFGHIKJKLMNOPQRSTUVWXYZ', '')
+math.randomseed(os.time())
+local function generate_random_string(len)
+  local result = ''
+  for _ = 1, len do
+    result = result .. chars[math.random(1, #chars)]
+  end
+  return result
+end
+
+local function make_completion_function(options)
+  local id = generate_random_string(5)
```

Comment:
Why can't we just use an incrementing id instead of generating a random string?

Reply:
good point well made :D no idea why i went straight to random string there, especially with how cumbersome it is in lua